pkg/vsphere/vm/vm_test.go

```diff
@@ -436,3 +446,145 @@ func TestBfsSnapshotTree(t *testing.T) {
 		t.Errorf("Should not found snapshot")
 	}
 }
+
+// TestProperties test vm.properties happy path and fix vm path
+func TestProperties(t *testing.T) {
+	ctx := context.Background()
+
+	// Nothing VC specific in this test, so we use the simpler ESX model
+	model := simulator.ESX()
+	defer model.Remove()
+	err := model.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	server := model.Service.NewServer()
+	defer server.Close()
+
+	client, err := govmomi.NewClient(ctx, server.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Any VM will do
+	finder := find.NewFinder(client.Client, false)
+	vmo, err := finder.VirtualMachine(ctx, "/ha-datacenter/vm/*_VM0")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	config := &session.Config{
+		Service:        server.URL.String(),
+		Insecure:       true,
+		Keepalive:      time.Duration(5) * time.Minute,
+		DatacenterPath: "",
+		DatastorePath:  "/ha-datacenter/datastore/*",
+		HostPath:       "/ha-datacenter/host/*/*",
+		PoolPath:       "/ha-datacenter/host/*/Resources",
+	}
+
+	s, err := session.NewSession(config).Create(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	vmm := NewVirtualMachine(ctx, s, vmo.Reference())
+	// Test the success path
+	var o mo.VirtualMachine
+	err = vmm.Properties(ctx, vmo.Reference(), []string{"config", "summary", "resourcePool", "parentVApp"}, &o)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//	// Inject invalid connection state to vm
+	ref := simulator.Map.Get(vmo.Reference()).(*simulator.VirtualMachine)
+	ref.Summary.Config.VmPathName = "[LocalDS_0] ha-host_VM0/ha-host_VM0.vmx"
```

Comment:
Looks like we need to set that field in vcsim.  You can also use this for now, rather than the hardcoded string:
`ref.Config.Files.VmPathName`